### PR TITLE
Cif 247 increase unit test coverage for cif

### PR DESCRIPTION
**Pull request by kcartier-wri**
Further increasing test coverage.

### CODE REVIEW

**Review by chrowe — COMMENTED** [review]
There is a lot to review here. I gave it a quick pass and all seams reasonable but I don't know if it is worth reviewing every test one by one to check that it is actually testing what I think it should be. 

I also ran `pytest` locally but it failed to run and when I tried to update my environment it failed because `exactextract` was not available. Maybe worth getting Github Actions to run using Conda?

**Comment by kcartier-wri** [comment]
I'm now testing how to create GitHub Actions workflow using Conda.

**Review by chrowe — APPROVED** [review]
I tried `conda env update -f environment.yml --prune` on my existing `cities-cif` Conda environment but it failed.
Then I tried creating a new Conda environment and that built fine.

All tests passed (except the 2 skipped ones). Why not just remove these since the layers are deprecated?

File: tests/test_layer_dimensions.py [hunk 1]
```diff
@@ -1,19 +1,128 @@
...
+    data = get_image_collection(ic, BBOX, 100, "test")
+    expected_x_dimension = 19
+    expected_y_dimension = 20
+    assert data.dims == {"x": expected_x_dimension, "y": expected_y_dimension}
+
+def test_albedo_dimensions():
+    data = Albedo().get_data(BBOX)
+    analysis_data = post_process_layer(data, value_threshold=0.1, convert_to_percentage=True)
+
+    expected_min = 0
```
Comment (jterry64) [inline tests/test_layer_dimensions.py:38]: Maybe include some info on how you got these expected values? Assuming you just manually tested it, but if we change the methodology or data sources in the future, someone may need to do that again.

**Review by jterry64 — CHANGES_REQUESTED** [review]
Some minor comments to increase clarity

File: tests/test_layer_dimensions.py [hunk 2]
```diff
@@ -1,19 +1,128 @@
...
+
+def test_tree_cover():
+    actual = TreeCover().get_data(BBOX).mean()
+    expected = 54.0
+    tolerance = 0.1
+    assert (
+            pytest.approx(expected, rel=tolerance) == actual
+    )
+
+def get_value_percent(data, value, full_count, precision):
```
[the remark this answers is not in the document]
Comment (kcartier-wri) [reply tests/test_layer_dimensions.py:111]: For simplicity, I removed the function and related tests. These are the tests that needed the most attention and I plan to improve further for https://gfw.atlassian.net/browse/CIF-280

File: tests/test_layer_dimensions.py [hunk 3]
```diff
@@ -1,19 +1,128 @@
...
+    )
+
+def test_read_image_collection_scale():
+    ic = ee.ImageCollection("ESA/WorldCover/v100")
+    data = get_image_collection(ic, BBOX, 100, "test")
+    expected_x_dimension = 19
+    expected_y_dimension = 20
+    assert data.dims == {"x": expected_x_dimension, "y": expected_y_dimension}
+
+def test_albedo_dimensions():
```
[the remark this answers is not in the document]
Comment (kcartier-wri) [reply tests/test_layer_dimensions.py:34]: "Dimensions" was abstract and I've renamed. I also renamed the overall file to "test_layer_metrics" and added a description near top of file.

File: tests/test_layer_dimensions.py (hunk 1 above)
Comment (kcartier-wri) [reply tests/test_layer_dimensions.py:38]: We don't actually post-process the data, so I've removed this step.

**Comment by kcartier-wri** [comment]
@jterry64 - Good points about the previous test_layer_dimensions.py (now test_layer_metrics) file. It is on my list of things to improve, so thank you for highlighting the file.
It may also be a good idea to include thest individual tests in the test_layers file in order to reduce testing run time.

**Review by jterry64 — APPROVED** [review]

**Pull request merged**